app/pages/learn/01_tutorial/08_more-resources/02_debugging.md

```diff
@@ -0,0 +1,136 @@
+3. Tracing the path of the code - When stopping at a breakpoint, the debugger shows what classes/methods were called in order to get there. You can even click on them to see what the variables in scope at those points are.
+
+<a id="basic">&nbsp;</a>
+## Debugger Basics
+
+There are four basic debugger commands to control the flow of execution once the debugger stops at your first breakpoint. For each of these commands, we will use the Flow class as an example.
+
+```java
+public class Flow {
+    public static void main(String args[]) {
+      System.out.println(debugging());
+    }
+    
+    public static int debugging() {
+        int num = investigate();
+        num++;
+        return num;
+    }
+    
+    public static int investigate() {
+        int found = 5;
+        return found;
+    }
+}
+```
+
+1. Step into - Tells the program to execute, but only to the first line of the method call. Suppose I have a breakpoint on line 7. When I tell the debugger to "step into", it goes to line 13.
+2. Step over - Tells the program to execute but not stop in any methods. If I have a breakpoint on line 7 and tell the debugger to 'step over", the debugger will then be on line 8. Choosing "step over" again will bring the debugger to line 9.
+3. Step out/return - Tells the program to run to the end of the method and go back to the caller. If I have a breakpoint on line 13 and choose "step out" or "step return", the debugger will be on line 7 with the result from the method call. (Step out and step return are the same thing. Different IDEs use different names.)
+4. Resume - Tells the program to keep going until it hits another breakpoint or completes.
+
+<a id="advanced">&nbsp;</a>
+## Advanced Techniques
+
+Debuggers have many advanced techniques. Three common ones are:
+
+1. Conditional breakpoint - Normally, the debugger stops where you asked for a breakpoint. If you are in a loop or have a clue what values trigger the problem, you don't want that. A conditional breakpoint allows you to add a bit of Java code to your breakpoint so it will only stop when that condition is true. This approach avoids having to hit resume a lot of times until you get to the value you care about.
+2. Evaluation - Once you get to your breakpoint, you can write Java code to determine the state of affairs. For example, you can call methods on the available variables.
+3. Hot replacement - You can manually change the value of a variable in the debugger and let the code continue to run. It will use your new, corrected value instead of the original one. This lets you explore the impact of a potential fix.
+
+<a id="docs">&nbsp;</a>
+## Documentation
+
+Now that you know the concepts of using a debugger, it is time to look at the documentation for your IDE! Pay attention to the keyboard shortcuts and where each button is located.
+
+1. [Eclipse](https://www.eclipse.org/community/eclipse_newsletter/2017/june/article1.php)
```

Comment:
I think most links in this repository are using [a JSON file in `app/data`](https://github.com/java/devjava-content/tree/main/app/data), e.g. https://github.com/java/devjava-content/blob/e8828c7cd05feb77fcebc0037be2058ba2fead4d/app/pages/learn/01_tutorial/01_your-first-java-app/01_getting-started-with-java.md?plain=1#L492-L494

Reply:
The file named javadoc.json? I agree there is non javadoc in there, but the name is misleading. When I wrote my other (cert) article, I started with the links in there and then wound up inlining them to the file. So I'm thinking I shouldn't move my links to a file named javadoc unless there is clarity in the readme/a  possible rename that all links should go there.

Reply:
Resolving because Billy didn't take issue with the links